chain-factory/src/factory.rs

```diff
@@ -62,18 +104,177 @@ pub trait FactoryModule {
+            .from_source(source_address)
+            .code_metadata(metadata)
+            .arguments_raw(args)
+            .returns(ReturnsNewManagedAddress)
+            .sync_call();
+
+        self.all_deployed_contracts(chain_id)
+            .insert(ContractMapArgs {
+                id: ScArray::SovereignHeaderVerifier,
+                address: header_verifier_address,
+            });
+    }
+
+    #[only_admin]
+    #[endpoint(deployCrossChainOperation)]
+    fn deploy_cross_chain_operation(
+        &self,
+        chain_id: ManagedBuffer,
+        is_sovereign_chain: bool,
+        opt_wegld_identifier: Option<TokenIdentifier>,
+        opt_sov_token_prefix: Option<ManagedBuffer>,
+    ) {
+        let source_address = self.cross_chain_operations_template().get();
+        let token_handler_address = self.token_handler_template().get();
+
+        let metadata =
+            CodeMetadata::PAYABLE_BY_SC | CodeMetadata::UPGRADEABLE | CodeMetadata::READABLE;
+
+        let mut args = ManagedArgBuffer::new();
+        args.push_arg(is_sovereign_chain);
+        args.push_arg(token_handler_address);
+        args.push_arg(opt_wegld_identifier);
+        args.push_arg(opt_sov_token_prefix);
+
+        let cross_chain_operations_address = self
+            .tx()
+            .raw_deploy()
+            .from_source(source_address)
+            .code_metadata(metadata)
+            .arguments_raw(args)
+            .returns(ReturnsNewManagedAddress)
+            .sync_call();
+
+        self.all_deployed_contracts(chain_id)
+            .insert(ContractMapArgs {
+                id: ScArray::SovereignCrossChainOperation,
+                address: cross_chain_operations_address,
+            });
+    }
+
+    #[only_admin]
+    #[endpoint(deployFeeMarket)]
+    fn deploy_fee_market(
+        &self,
+        chain_id: ManagedBuffer,
+        esdt_safe_address: ManagedAddress,
+        price_aggregator_address: ManagedAddress,
+    ) {
+        let source_address = self.fee_market_template().get();
+
+        let metadata =
+            CodeMetadata::PAYABLE_BY_SC | CodeMetadata::UPGRADEABLE | CodeMetadata::READABLE;
+
+        let mut args = ManagedArgBuffer::new();
+        args.push_arg(esdt_safe_address);
+        args.push_arg(price_aggregator_address);
+
+        let fee_market_address = self
+            .tx()
+            .raw_deploy()
+            .from_source(source_address)
+            .code_metadata(metadata)
+            .arguments_raw(args)
+            .returns(ReturnsNewManagedAddress)
+            .sync_call();
+
+        self.all_deployed_contracts(chain_id)
+            .insert(ContractMapArgs {
+                id: ScArray::FeeMarket,
+                address: fee_market_address,
+            });
+    }
+
+    #[only_admin]
+    #[endpoint(deployTokenHandler)]
+    fn deploy_token_handler(&self, chain_id: ManagedBuffer) {
+        let source_address = self.fee_market_template().get();
+
+        let metadata =
```

Comment:
Use `self.blockchain().get_code_metadata(&source_address)` to get metadata.

Reply:
thank you, didn't know about that, I will add it nou